computer_vision_draft/houghTransform/houghTransform/main.cpp

```diff
@@ -0,0 +1,106 @@
+#include <opencv2/opencv.hpp>
+#include <iostream>
+#include <string>
+#include <fstream>
+#include<windows.h>  
+#include<chrono>
+#include<cmath>
+#include "DrivabilityDetector.h"
+
+using namespace cv;
+using namespace std;
+
+//Function for rotating 
+static Mat rotateImage(Mat image, double angle);
+
+//For measuring elapsed time
+template <
+	class result_t = std::chrono::milliseconds,
+	class clock_t = std::chrono::steady_clock,
+	class duration_t = std::chrono::milliseconds
+>
+auto since(std::chrono::time_point<clock_t, duration_t> const& start)
+{
+	return std::chrono::duration_cast<result_t>(clock_t::now() - start);
+}
+
+
+//Main loop
+int main(int argc, char** argv)
+{
+	//Vector for storing filenames
+	vector<cv::String> fn;
+	cv::glob("capture4/*.png", fn, true);
+	if (fn.size() == 0) {
+		cout << "No files in folder" << endl;
+		return -1;
+	}
+
+	//Open windows for showing progress
+	String windowName = "Drivability map"; 
+	namedWindow(windowName); 
+	String windowName2 = "Lines";
+	namedWindow(windowName2); 
+
+	Point2i centerPoint(100, 100); //Intial guess for center point(in scaled coordinates)
+	DrivabilityDetector test(0.8, centerPoint,5); //Create an object of type "DrivabilityDetector", with past weight 0.8, initial guess centerPoint and a 5 point moving average filter
+	for (size_t i = 0;i < fn.size();i++)
+	{
+		//Read image
+		auto start = std::chrono::steady_clock::now(); //Start time meas
+		Mat image = imread(fn[i]); // Read image
+		if (image.empty()) // Check for failure
+		{
+			cout << "Could not open or find the image" << endl;
+			system("pause"); //wait for any key press
+			return -1;
+		}
+
+		//Process image
+		Mat grayImage;
+		cvtColor(image, grayImage, COLOR_RGB2GRAY);			//GRAYSCALE
+		cv::flip(grayImage, grayImage, 0);					//Flip image if camera was upside down
+		float scale = 0.5;									//Scale down image
+		resize(grayImage, grayImage, Size(), scale, scale); // Resize
+		grayImage = rotateImage(grayImage, (double)3);		//Correct tilt of camera
+		
+		auto preprocess = since(start).count();									
+		std::cout << "Preprocessing complete(ms) =" << preprocess;
+
+		//Run computer vision function
+		int rowFromBottom = test.calculateRowFromBottom(grayImage);
+
+		auto end = since(start).count() - preprocess;
+		std::cout << " Calculations complete(ms) =" << end << std::endl;
+		
+		cout << "Distance from bottom : " << test.getRowFromBottom() << endl;
+
+		//Display images
+		Mat overlayedImage;
+		Mat testImage = test.getDrivabilityMap();
+		addWeighted(grayImage, 1, test.getDrivabilityMap(), 0.5, 0, overlayedImage);
```

Comment:
Hide the remaining code here behind some `ifndef RACE`

Reply:
Applied suggestion